operator/v1/types_machineconfiguration.go

```diff
@@ -158,3 +179,205 @@ const (
+	Units []NodeDisruptionPolicyUnit `json:"units,omitempty"`
+	// sshkey is the overall sshkey MachineConfig definition
+	// +optional
+	SSHKey NodeDisruptionPolicySSHKey `json:"sshkey,omitempty"`
+}
+
+// NodeDisruptionPolicyFile is a file entry and corresponding actions to take
+type NodeDisruptionPolicyFile struct {
+	// path is the location of the file being managed through a MachineConfig.
+	// Actions specified will be applied when changes to the file at the path
+	// configured in this field.
+	// +kubebuilder:validation:Required
+	Path string `json:"path"`
+	// actions represents the series of commands to be executed on changes to the file at
+	// the corresponding file path. Actions will be applied in the order that
+	// they are set in this list. If there are other incoming changes to other MachineConfig
+	// entries in the same update that require a reboot, the reboot will supercede these actions.
+	// Valid actions are Reboot, Drain, Reload, DaemonReload and None.
+	// The Reboot action and the None action cannot be used in conjunction with any of the other actions.
+	// This list supports a maximum of 10 entries.
+	// +kubebuilder:validation:Required
+	// +listType=atomic
+	// +kubebuilder:validation:MaxItems=10
+	// +kubebuilder:validation:XValidation:rule="self.exists(x, x.type=='Reboot') ? size(self) == 1 : true", message="Reboot action can only be specified standalone, as it will override any other actions"
+	// +kubebuilder:validation:XValidation:rule="self.exists(x, x.type=='None') ? size(self) == 1 : true", message="None action can only be specified standalone, as it will override any other actions"
+	Actions []NodeDisruptionPolicyAction `json:"actions"`
+}
+
+// NodeDisruptionPolicyUnit is a systemd unit name and corresponding actions to take
+type NodeDisruptionPolicyUnit struct {
+	// name represents the service name of a systemd service managed through a MachineConfig
+	// Actions specified will be applied for changes to the named service.
+	// Service names should be of the format ${NAME}.${SERVICETYPE} and can up to 30 characters long.
+	// ${NAME} should be atleast one character long and can consist of uppercase letters, lowercase letters, digits, underscores, or hyphens.
+	// ${SERVICETYPE} must be ".service", ".socket", ".device", ".mount", ".automount", ".swap", ".target", ".path", ".timer",
+	// ".snapshot", ".slice" or ".scope".
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MaxLength=30
+	// +kubebuilder:validation:Pattern:=`^[a-zA-Z0-9_-]+\.(service|socket|device|mount|automount|swap|target|path|timer|snapshot|slice|scope)$`
+	Name string `json:"name"`
+
+	// Name regex breakdown:
+	// [a-zA-Z0-9_-]+ : Matches one or more characters consisting of uppercase letters, lowercase letters, digits, underscores, or hyphens.
+	// \. : Matches a period.
+	// (service|socket|device|mount|automount|swap|target|path|timer|snapshot|slice|scope) : Matches any of the specified service types.
```

Comment:
Maybe do this before the field, rather than after. Also please check that none of the generated docs or openapi include this text